ush/forecast_postdet.sh

```diff
@@ -350,27 +349,52 @@ WW3_postdet() {
     ${NCP} "${ww3_restart_file}" "${DATA}/restart.ww3" \
-      || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
+          || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
     first_ww3_restart_out=$(date --utc -d "${restart_date:0:8} ${restart_date:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    if [[ "${DOIAU:-NO}" == "YES" ]]; then
+      first_ww3_restart_out=$(date --utc -d "${first_ww3_restart_out:0:8} ${first_ww3_restart_out:8:2} + ${half_window} hours" +%Y%m%d%H)
+    fi 
   else
-    if [[ "${RERUN}" == "YES" ]]; then
-      # In the case of a RERUN, the WW3 restart file is required
-      echo "FATAL ERROR: WW3 restart file '${ww3_restart_file}' not found for RERUN='${RERUN}', ABORT!"
-      exit 1
-    else
-      echo "WARNING: WW3 restart file '${ww3_restart_file}' not found for warm_start='${warm_start}', will start from rest!"
-      first_ww3_restart_out=${model_start_date_current_cycle}
+    # now check to see if a NetCDF restart exists: 
+    seconds=$(to_seconds "${restart_date:8:2}0000")  # convert HHMMSS to seconds
+    ww3_restart_dest_file="ufs.cpld.ww3.r.${restart_date:0:4}-${restart_date:4:2}-${restart_date:6:2}-${seconds}.nc"
+
+    ww3_restart_file="${restart_dir}/${restart_date:0:8}.${restart_date:8:2}0000.restart.ww3.nc"
+    if [[ -s "${ww3_restart_file}" ]]; then  
+      # NetCDF restart exists: 
+      export WW3_restart_from_binary=false
+      ${NCP} "${ww3_restart_file}" "${DATA}/${ww3_restart_dest_file}" \
+             || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
+      first_ww3_restart_out=$(date --utc -d "${restart_date:0:8} ${restart_date:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+      if [[ "${DOIAU:-NO}" == "YES" ]]; then
+        first_ww3_restart_out=$(date --utc -d "${first_ww3_restart_out:0:8} ${first_ww3_restart_out:8:2} + ${half_window} hours" +%Y%m%d%H)
+      fi
+    else    
+      if [[ "${RERUN}" == "YES" ]]; then
+        # In the case of a RERUN, the WW3 restart file is required
+        echo "FATAL ERROR: WW3 restart file '${ww3_restart_file}' not found for RERUN='${RERUN}', ABORT!"
+        exit 1
+      else
+        # TODO: See if this will work still with new options... might not
+        echo "WARNING: WW3 restart file '${ww3_restart_file}' not found for warm_start='${warm_start}', will start from rest!"
+        first_ww3_restart_out=${model_start_date_current_cycle}
+      fi
     fi
   fi
 
   # Link restart files
   for (( vdate = first_ww3_restart_out; vdate <= forecast_end_cycle;
          vdate = $(date --utc -d "${vdate:0:8} ${vdate:8:2} + ${restart_interval} hours" +%Y%m%d%H) )); do
-    ww3_restart_file="${vdate:0:8}.${vdate:8:2}0000.restart.ww3"
-    ${NLN} "${DATArestart}/WW3_RESTART/${ww3_restart_file}" "${ww3_restart_file}"
+	  seconds=$(to_seconds "${vdate:8:2}0000")  # convert HHMMSS to seconds
+    ww3_restart_ufs_file="ufs.cpld.ww3.r.${vdate:0:4}-${vdate:4:2}-${vdate:6:2}-${seconds}.nc"
```

Comment:
@correcting the indentation (editor on GH might not result is the correct suggestion)
```suggestion
    seconds=$(to_seconds "${vdate:8:2}0000")  # convert HHMMSS to seconds
    ww3_restart_ufs_file="ufs.cpld.ww3.r.${vdate:0:4}-${vdate:4:2}-${vdate:6:2}-${seconds}.nc"
```